src/app/dashboard/grades/page.tsx

```diff
@@ -0,0 +1,190 @@
+'use client';
+
+import type { ReactElement } from 'react';
+
+import Subject from '@/components/grades/subject';
+
+export type Grade = {
+  value: number;
+  description: string;
+  category:
+    | 'Kartkówka'
+    | 'Sprawdzian'
+    | 'Odpowiedź ustna'
+    | 'Projekt'
+    | 'Laboratorium'
+    | 'Prezentacja'
+    | 'Zadanie domowe'
+    | 'Wypracowanie'
+    | 'Praca domowa';
+};
+
+export type SubjectGrades = {
+  subject: string;
```

Comment:
```suggestion
  subject: uuid :: subject
```
1. Subjects should be UUIDs not hardcoded texts...



also I think docstruct of:


**a**:
```
<student>: {
  subjects: {
    <subject>: {
      grades: {
        <grade>: {...}
      }
    ...
    }
  }
  ...
}
```


**b**:
```
<student>: {
  grades: {
    <grade>: {
      subject: <subject>
      ...
    }
  }
}
```


ALTERNATIVELY we could go back to relational, where grade would be:
```
key <grade> UUID
rel <student> UUID 
-> rel <school> UUID
rel <subject> UUID
-> rel <school> UUID
rel <teacher> UUID
-> rel <school> UUID
data ...
```


**PLEASE**  
Do verify that *AI-gen* code for mockups/data is sensible
Try always picturing the structures (in multiple implementations, like in my example above) and include them in comments 😃